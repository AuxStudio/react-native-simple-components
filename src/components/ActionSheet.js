import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import styleConstants from "../assets/styleConstants";

import { Touchable, IconTextRow } from "react-native-simple-components";
import { AnimateTranslateY } from "react-native-simple-animators";

export default class ActionSheet extends React.Component {
    constructor(props) {
        super(props);

        this.handleSelect = this.handleSelect.bind(this);

        this.state = {
            shouldAnimateOut: false,
        };
    }

    static get propTypes() {
        return {
            options: PropTypes.array,
            handlePress: PropTypes.func,
            rowHeight: PropTypes.func, // needed for animation
            textStyle: PropTypes.any,
            iconStyle: PropTypes.any,
            rowStyle: PropTypes.any,
            style: PropTypes.any,
        };
    }

    handleSelect(item) {
        this.setState({
            shouldAnimateOut: true,
        });
        setTimeout(() => {
            this.props.handlePress && this.props.handlePress(item);
        }, 500);
    }

    render() {
        const ROW_HEIGHT = this.props.rowHeight ? this.props.rowHeight : 56;

        return (
            <AnimateTranslateY
                initialValue={(this.props.options.length + 1) * ROW_HEIGHT}
                finalValue={0}
                shouldAnimateIn={!this.state.shouldAnimateOut}
                shouldAnimateOut={this.state.shouldAnimateOut}
                style={[styles.container, this.props.style]}>
                {this.props.options.map(item => {
                    return (
                        <Touchable
                            onPress={() => this.handleSelect(item.text)}
                            style={styles.row}
                            key={item.text}>
                            <IconTextRow
                                iconName={item.iconName}
                                iconStyle={[styles.icon, this.props.iconStyle]}
                                text={item.text}
                                textStyle={[styles.text, this.props.textStyle]}
                                style={[styles.row, { height: ROW_HEIGHT }]}
                            />
                        </Touchable>
                    );
                })}
                <Touchable
                    onPress={() => this.handleSelect("Cancel")}
                    style={styles.row}>
                    <IconTextRow
                        iconName="cancel"
                        iconStyle={[styles.icon, this.props.iconStyle]}
                        text="Cancel"
                        textStyle={[styles.text, this.props.textStyle]}
                        style={styles.row}
                    />
                </Touchable>
            </AnimateTranslateY>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        position: "absolute",
        bottom: 0,
        left: 0,
        right: 0,
        zIndex: 100,
        elevation: 100,
        backgroundColor: styleConstants.white,
    },
    row: {
        paddingHorizontal: 8,
        borderBottomWidth: 1,
        borderBottomColor: styleConstants.dividerColor,
        alignItems: "center",
    },
    text: {
        fontSize: styleConstants.regularFont,
        color: styleConstants.primaryText,
        ...styleConstants.primaryFont,
        textAlign: "left",
    },
    icon: {
        fontSize: styleConstants.iconFont,
        color: styleConstants.secondaryText,
        marginRight: 8,
    },
});
