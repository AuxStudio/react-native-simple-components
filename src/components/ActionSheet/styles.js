import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    zIndex: 100,
    elevation: 100,
    backgroundColor: styleConstants.colors.white,
  },
  row: {
    paddingHorizontal: styleConstants.dimensions.padding.small,
    alignItems: 'center',
  },
  text: {
    fontSize: styleConstants.fonts.sizes.regular,
    color: styleConstants.colors.primaryText,
    textAlign: 'left',
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.primaryText,
    marginRight: styleConstants.dimensions.padding.small,
  },
});

export default styles;
