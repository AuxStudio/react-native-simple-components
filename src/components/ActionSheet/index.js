import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes, Text } from 'react-native';
import Animator from 'react-native-simple-animators';

import styles from './styles';
import Touchable from '../Touchable';
import IconTextRow from '../IconTextRow';

export default class ActionSheet extends React.Component {
  constructor(props) {
    super(props);

    this.handleSelect = this.handleSelect.bind(this);

    this.state = {
      shouldAnimateOut: false,
    };
  }

  static get propTypes() {
    return {
      options: PropTypes.arrayOf(
        PropTypes.shape({ iconName: PropTypes.string, text: PropTypes.string }),
      ),
      handlePress: PropTypes.func,
      rowHeight: PropTypes.number,
      textStyle: Text.propTypes.style,
      iconStyle: Text.propTypes.style,
      style: ViewPropTypes.style,
    };
  }

  static defaultProps = {
    options: [{ iconName: 'check', text: 'Option 1' }],
    rowHeight: 56,
  };

  handleSelect(item) {
    this.setState({
      shouldAnimateOut: true,
    });
    setTimeout(() => {
      if (this.props.handlePress) {
        this.props.handlePress(item);
      }
    }, 500);
  }

  render() {
    const initialValue = (this.props.options.length + 1) * this.props.rowHeight;

    return (
      <Animator
        type="translateY"
        initialValue={initialValue}
        finalValue={0}
        shouldAnimateIn={!this.state.shouldAnimateOut}
        shouldAnimateOut={this.state.shouldAnimateOut}
        style={[styles.container, this.props.style]}
      >
        {this.props.options.map((item) => {
          return (
            <Touchable
              onPress={() => this.handleSelect(item.text)}
              style={styles.itemContainer}
              key={item.text}
            >
              <IconTextRow
                iconName={item.iconName}
                iconStyle={[styles.icon, this.props.iconStyle]}
                text={item.text}
                textStyle={[styles.text, this.props.textStyle]}
                style={[styles.itemContainer, { height: this.props.rowHeight }]}
              />
            </Touchable>
          );
        })}
        <Touchable onPress={() => this.handleSelect('Cancel')} style={styles.itemContainer}>
          <IconTextRow
            iconName="cancel"
            iconStyle={[styles.icon, this.props.iconStyle]}
            text="Cancel"
            textStyle={[styles.text, this.props.textStyle]}
            style={[styles.itemContainer, { height: this.props.rowHeight }]}
          />
        </Touchable>
      </Animator>
    );
  }
}
