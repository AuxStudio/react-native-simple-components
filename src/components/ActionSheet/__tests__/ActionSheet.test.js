import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-native-simple-animators', () => ({
  AnimateTranslateY: 'AnimateTranslateY',
}));

import ActionSheet from '../'; // eslint-disable-line

describe('ActionSheet', () => {
  it('renders with all props', () => {
    const component = renderer.create(
      <ActionSheet
        options={[{ iconName: 'check', text: 'Test 1' }, { iconName: 'close', text: 'Test 2' }]}
        handlePress={jest.fn()}
        rowHeight={100}
        textStyle={{ color: 'red' }}
        iconStyle={{ color: 'red' }}
        style={{ backgroundColor: 'blue' }}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<ActionSheet />)).toMatchSnapshot();
  });

  // it('renders and selects a menu item', () => {
  //   const component = renderer.create(
  //     <ActionSheet
  //       options={[{ iconName: 'check', text: 'Test 1' }, { iconName: 'close', text: 'Test 2' }]}
  //       handlePress={jest.fn()}
  //       rowHeight={100}
  //       textStyle={{ color: 'red' }}
  //       iconStyle={{ color: 'red' }}
  //       style={{ backgroundColor: 'blue' }}
  //     />,
  //   );
  //   const instance = component.getInstance();

  //   instance.handleSelect({ iconName: 'check', text: 'Test 1' });
  //   expect(instance.state.shouldAnimateOut).toBe(true);
  // });
});
