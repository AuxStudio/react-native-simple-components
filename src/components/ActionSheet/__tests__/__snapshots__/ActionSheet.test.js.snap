// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet renders and selects a menu item 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "iconName": "check",
        "text": "Test 1",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`ActionSheet renders with all props 1`] = `
<Animator
  finalValue={0}
  initialValue={300}
  shouldAnimateIn={true}
  shouldAnimateOut={false}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "elevation": 100,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "zIndex": 100,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
  type="translateY"
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 24,
              },
              Array [
                Object {
                  "color": "#212121",
                  "fontSize": 24,
                  "marginRight": 8,
                },
                Object {
                  "color": "red",
                },
              ],
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#212121",
              "flex": 1,
              "fontSize": 17,
              "textAlign": "right",
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 17,
                "textAlign": "left",
              },
              Object {
                "color": "red",
              },
            ],
          ]
        }
      >
        Test 1
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 24,
              },
              Array [
                Object {
                  "color": "#212121",
                  "fontSize": 24,
                  "marginRight": 8,
                },
                Object {
                  "color": "red",
                },
              ],
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#212121",
              "flex": 1,
              "fontSize": 17,
              "textAlign": "right",
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 17,
                "textAlign": "left",
              },
              Object {
                "color": "red",
              },
            ],
          ]
        }
      >
        Test 2
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 24,
              },
              Array [
                Object {
                  "color": "#212121",
                  "fontSize": 24,
                  "marginRight": 8,
                },
                Object {
                  "color": "red",
                },
              ],
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#212121",
              "flex": 1,
              "fontSize": 17,
              "textAlign": "right",
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 17,
                "textAlign": "left",
              },
              Object {
                "color": "red",
              },
            ],
          ]
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</Animator>
`;

exports[`ActionSheet renders with minimum required props 1`] = `
<Animator
  finalValue={0}
  initialValue={112}
  shouldAnimateIn={true}
  shouldAnimateOut={false}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "elevation": 100,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "zIndex": 100,
      },
      undefined,
    ]
  }
  type="translateY"
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 56,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 24,
              },
              Array [
                Object {
                  "color": "#212121",
                  "fontSize": 24,
                  "marginRight": 8,
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#212121",
              "flex": 1,
              "fontSize": 17,
              "textAlign": "right",
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 17,
                "textAlign": "left",
              },
              undefined,
            ],
          ]
        }
      >
        Option 1
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "opacity": 1,
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#E0E0E0",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 56,
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 24,
              },
              Array [
                Object {
                  "color": "#212121",
                  "fontSize": 24,
                  "marginRight": 8,
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#212121",
              "flex": 1,
              "fontSize": 17,
              "textAlign": "right",
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 17,
                "textAlign": "left",
              },
              undefined,
            ],
          ]
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</Animator>
`;
